### ISSUE TITLE
Using `shadow=True` does not allow for white background in `ax.legend`

### ISSUE THREAD

**Issue by kdheepak** [matplotlib/matplotlib#8943]
### Bug report

**Bug summary**

Using `shadow=True` does not allow for white background in `legend`

**Code for reproduction**

```python
ax.legend(shadow=True, facecolor='white')
```

**Actual outcome**

![screen shot 2017-07-26 at 16 29 06](https://user-images.githubusercontent.com/1813121/28646542-6dfc0584-721f-11e7-8479-eecc341ac0da.png)

**Expected outcome**

![screen shot 2017-07-26 at 16 29 36](https://user-images.githubusercontent.com/1813121/28646552-744a75b0-721f-11e7-8774-06fa517efe94.png)

**Matplotlib version**

  * Operating System: Mac
  * Matplotlib Version: 2.0.2
  * Python Version: 3.6

**Comment by tacaswell** [comment]
The issue is that is 2.0 we made the default alpha of the legend <1 (see http://matplotlib.org/users/dflt_style_changes.html#legends ) so you see the shadow box blended through.  If you pass `framealpha=1` you will get the expected behavior.

**Comment by tacaswell** [comment]
```python
import matplotlib
matplotlib.use('qt5Agg')
import matplotlib.pyplot as plt
plt.ion()

fig, (ax, ax2) = plt.subplots(1, 2)

ax.plot(range(5), label='broken')
ax2.plot(range(5), label='works')

ax.legend(shadow=True, facecolor='w')
ax2.legend(shadow=True, facecolor='w', framealpha=1)
```

![so](https://user-images.githubusercontent.com/199813/28698302-19a16080-7308-11e7-87b9-3161f85a5273.png)

Exact work:

 - do not consult the rcparams for framealpha if `shadow` is True in the legend `__init__` if not explicitly passed
 - add a test (checking the alpha of the legend patch is enough, do not need a image test)
 - document somewhat weird default behavior in docstring

**Comment by tacaswell** [comment]
Closed by #8988
